quickjs.c

```diff
@@ -21082,10 +20783,7 @@ static __exception int next_token(JSParseState *s)
     case '^':
         if (p[1] == '=') {
             p += 2;
-            if (s->cur_func->js_mode & JS_MODE_MATH)
-                s->token.val = TOK_MATH_POW_ASSIGN;
-            else
-                s->token.val = TOK_XOR_ASSIGN;
+            s->token.val = TOK_XOR_ASSIGN;
```

Comment:
I believe you can replace this block (the code inside the `#ifdef CONFIG_BIGNUM`) with just the `#else` case.